src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -759,20 +759,22 @@ private static AmazonWebServicesCredentials getCredentials(@Nullable String cred
     public AmazonEC2 connect() throws AmazonClientException {
         try {
             if (connection == null) {
-                connection = connect(createCredentialsProvider(), getEc2EndpointUrl());
+                synchronized (this) {
```

Comment:
```suggestion
                synchronized (this) {
                    if (connection == null)
```

Otherwise, you still end up with multiple threads calling the method.